app/models/site.rb

```diff
@@ -11,11 +11,27 @@ class Site < ActiveRecord::Base
            :announcement_text=, :marketing_text=, :featured_researcher=,
            to: :content_blocks
 
+  def content_by_name(name)
```

Comment:
@mjgiarlo I'd say it's a partial refactoring. Cloning the existing way of updating ContentBlocks requires a specific method to be defined so it can be passed through the Site via delegate. This is what I originally proposed in a sufia PR.

I could refactor the other content blocks to use this new content_by_name method, if you like?